### PR TITLE
Mention distributed profiling in documentation

### PR DESCRIPTION
**Pull request by rohan-varma**
Added a simple section indicating distributed profiling is expected to work similar to other torch operators, and is supported for all communication backends out of the box.

### CODE REVIEW

**Comment by rohan-varma** [comment]
prof_doc_ss 2

File: docs/source/distributed.rst [hunk 1]
```diff
@@ -428,6 +428,22 @@ Collective functions
...
+
+::
+
+    import torch
+    import torch.distributed as dist
+    with torch.profiler():
+        tensor = torch.randn(20, 10)
+        dist.all_reduce(tensor)
+
+Please refer to the `profiler documentation <https://pytorch.org/docs/stable/profiler.html>`__ for a full overview of profiler features.
```
Comment (SciPioneer) [inline docs/source/distributed.rst:445]: Shall we point to the webpage on the master branch instead?

File: docs/source/distributed.rst [hunk 2]
```diff
@@ -428,6 +428,22 @@ Collective functions
 
     :class:`~torch.distributed.ReduceOp` is recommended to use instead.
 
+Profiling Collective Communication
+-----------------------------------------
+
+Note that you can use ``torch.profiler`` or ``torch.autograd.profiler`` to profile collective communication and point-to-point communication APIs mentioned here. All out-of-the-box backends (``gloo``,
```
Comment (SciPioneer) [inline docs/source/distributed.rst:434]: IIRC, the latter `torch.autograd.profiler` probably will be deprecated in favor of `torch.profiler`, so we can mention this:

``torch.profiler`` (recommended, only available after 1.8.1)

**Review by wayi1 — APPROVED** [review]

File: docs/source/distributed.rst (hunk 2 above)
Comment (rohan-varma) [reply docs/source/distributed.rst:434]: Yes good point, will add a note

**Pull request closed without merging**